pkg/ami/ami_test.go

```diff
@@ -0,0 +1,83 @@
+package ami_test
+
+import (
+	"reflect"
+
+	"github.com/weaveworks/eksctl/pkg/ami"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AMI Resolution", func() {
+	type ResolveCase struct {
+		Region       string
+		InstanceType string
+		ExpectedAMI  string
+		ExpectError  bool
+	}
+
+	DescribeTable("When resolving an AMI to use",
+		func(c ResolveCase) {
+			actualAmi, err := ami.ResolveAMI(c.Region, c.InstanceType)
+			Expect(actualAmi).Should(Equal(c.ExpectedAMI))
+			if c.ExpectError {
+				Expect(err).Should(HaveOccurred())
+				Expect(err).Should(MatchError(ami.NewErrFailedAMIResolution(c.Region, c.InstanceType)))
+				errorType := reflect.TypeOf(err).Elem().Name()
+				Expect(errorType).To(Equal("ErrFailedAMIResolution"))
+			} else {
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+		},
+		Entry("with non-gpu instance and us-west-2", ResolveCase{
+			Region:       "us-west-2",
+			InstanceType: "t2.medium",
+			ExpectedAMI:  "ami-08cab282f9979fc7a",
+			ExpectError:  false,
+		}),
+		Entry("with non-gpu instance and us-east-1", ResolveCase{
+			Region:       "us-east-1",
+			InstanceType: "t2.medium",
+			ExpectedAMI:  "ami-0b2ae3c6bda8b5c06",
+			ExpectError:  false,
+		}),
+		Entry("with non-gpu instance and eu-west-1", ResolveCase{
+			Region:       "eu-west-1",
+			InstanceType: "t2.medium",
+			ExpectedAMI:  "ami-066110c1a7466949e",
+			ExpectError:  false,
+		}),
+		Entry("with non-gpu instance and non-eks enabled region", ResolveCase{
+			Region:       "ap-northeast-1",
+			InstanceType: "t2.medium",
+			ExpectedAMI:  "",
+			ExpectError:  true,
+		}),
+		Entry("with gpu (p2) instance and us-west-2", ResolveCase{
+			Region:       "us-west-2",
+			InstanceType: "p2.xlarge",
+			ExpectedAMI:  "ami-0d20f2404b9a1c4d1",
+			ExpectError:  false,
+		}),
+		Entry("with gpu (p3) instance and us-east-1", ResolveCase{
+			Region:       "us-east-1",
+			InstanceType: "p3.2xlarge",
+			ExpectedAMI:  "ami-09fe6fc9106bda972",
```

Comment:
It seems like we'd need to update this every time AMI changes... If we solve #49 by code generation, would you generate code for this also? I seems like a map would be easier to generate then switch+tests, or we could use a static JSON asset... I'm happy to land this as is, but just saying that we might need to change it soon enough.

Reply:
Yeah i completely agree. I'll take a look at doing this and then squash/rebase

Reply:
I'll work on #49 after this.